### PR TITLE
PORT-10037 Add a description for each property in the branch_protection blueprint

### PR DESCRIPTION
**Pull request by kodjomiles**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.
Added descriptions to branch_protection blueprint to give more clarity on the fucntion of the property

## Updated docs pages

Please also include the path for the updated docs

- Quickstart (`/`)
- Blueprint (`/platform-overview/port-components/blueprint`)
- /build-your-software-catalog/sync-data-to-catalog/git/github/examples/resource-mapping-examples/#map-repositories-and-branches

### CODE REVIEW

File: docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx [hunk 1]
```diff
@@ -11,67 +11,85 @@
       "require_approval_count": {
         "title": "Require approvals",
         "type": "number",
-        "icon": "DefaultProperty"
+        "icon": "DefaultProperty",
+        "description": "Specifies the number of approvals required before merging a pull request."
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx:15]:
```suggestion
        "description": "The number of approvals required before merging a pull request"
```

File: docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx [hunk 2]
```diff
@@ -11,67 +11,85 @@
...
       },
       "is_protected": {
         "title": "Is branch protected",
         "type": "boolean",
-        "icon": "DefaultProperty"
+        "icon": "DefaultProperty",
+        "description": "Indicates whether the branch is protected, enforcing certain rules before changes can be merged."
```
Comment (PeyGis) [inline docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx:21]:
```suggestion
        "description": "Whether the branch is protected, enforcing certain rules before changes can be merged"
```

**Review by PeyGis — APPROVED** [review]
Looks good. just apply the suggested changes. And remove the dot at the end of every description

File: docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx (hunk 2 above)
Comment (hadar-co) [reply docs/build-your-software-catalog/sync-data-to-catalog/git/github/examples/example-branch/_git_exporter_example_branch_protection_blueprint.mdx:21]: @jadenmyles @PeyGis Using the same words from the title does not clarify much, I would say:
"Indicates whether certain rules must be met before changes can be merged"

**Pull request merged**